tool/tsh/common/db_print.go

```diff
@@ -141,3 +141,28 @@ func formatDatabaseRolesForDB(database types.Database, accessChecker services.Ac
 	}
 	return ""
 }
+
+func shouldShowListDatabasesHint(cf *CLIConf, numRows int) bool {
+	selector := newDatabaseResourceSelectors(cf)
+
+	return numRows >= minNumRowsToShowListDatabasesHint &&
+		cf.command == "db ls" &&
+		cf.SearchKeywords == "" &&
+		selector.IsEmpty()
+}
+
+func maybeShowListDatabasesHint(cf *CLIConf, w io.Writer, numRows int) {
+	if !shouldShowListDatabasesHint(cf, numRows) {
+		return
+	}
+
+	fmt.Fprint(w, listDatabaseHint)
+}
+
+// minNumRowsToShowListDatabasesHint is an arbitrary number selected to show
+// filtering hint for `tsh db ls` command when too many databases are listed.
+const minNumRowsToShowListDatabasesHint = 20
+
+const listDatabaseHint = "" +
+	"hint: use 'tsh db ls --search foo,bar' to search keywords\n" +
+	"      use 'tsh db ls key1=value1,key2=value2' to filter databases by labels\n\n"
```

Comment:
I don't have strong opition but why not hide this behind `tsh db ls --help` command ? 

```suggestion
const listDatabaseHelp = `
...


Examples:
 Search by keywords: 
 $ tsh db ls --search foo,bar' 
 Filter Database by labels 
 $ tsh db ls key1=value1,key2=value2
`

dbList.Help(listDatabaseHelp)

Reply:
Good call on the usage.

The hint is more useful in my opinion, as I don't think people check the usage for `tsh db ls`, especially for existing users. I have no idea these flags exist for a very long time. And I often forget to use them too.

What do you think if I keep the hint but also add the examples in addition?

Reply:
added https://github.com/gravitational/teleport/pull/40671/commits/b64a7d9bc04d59fb04d25bf5fca417582608b114